[16.0][ADD] shopfloor_reception_helpdesk/mobile

rousseldenis opened the pull request:
Depends on:
* https://github.com/OCA/helpdesk/pull/875

rousseldenis committed:
[16.0][ADD] shopfloor_reception_helpdesk

rousseldenis committed:
[16.0][ADD] shopfloor_reception_helpdesk

rousseldenis committed:
[16.0][ADD] shopfloor_reception_helpdesk_mobile

rousseldenis committed:
[16.0][ADD] shopfloor_reception_helpdesk_mobile

rousseldenis committed:
[DONT MERGE] test-requirements.txt

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk: Pass the data if user is allowed to create tickets

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk: Pass the data if user is allowed to create tickets

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk_mobile: Don't display Helpdesk button if not allowed

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk_mobile: Don't display Helpdesk button if not allowed

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk: Add the wizard as parameter to keep data

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk: Add the wizard as parameter to keep data

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk_mobile: Use wizard to keep data

rousseldenis committed:
[IMP] shopfloor_reception_helpdesk_mobile: Use wizard to keep data

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk: enables to specify the motive when creating a ticket.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk: enables to specify the motive when creating a ticket.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: add drop-down menu in UI to select motive when creating a ticket.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: add drop-down menu in UI to select motive when creating a ticket.

nicolas-delbovier-acsone commented:
Here is what I added in the UI for selecting a motive: 
<img width="803" height="845" alt="image" src="https://github.com/user-attachments/assets/1227b53a-89f6-4e5d-a2b4-531c79492ff8" />

Let me now what you think.

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdesk: also return motives not associated to any particular team in available motives during "start_helpdesk"

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdesk: also return motives not associated to any particular team in available motives during "start_helpdesk"

nicolas-delbovier-acsone committed:
[REF] shopfloor_reception_helpdesk: Adapt to changes in helpdesk_mgmt_stock dependency

Renaming wizard mehtod: `create_helpdesk_ticket` -> `action_create_helpdesk_ticket`

See PR: https://github.com/OCA/helpdesk/pull/875

nicolas-delbovier-acsone committed:
[REF] shopfloor_reception_helpdesk: Adapt to changes in helpdesk_mgmt_stock dependency

Renaming wizard mehtod: `create_helpdesk_ticket` -> `action_create_helpdesk_ticket`

See PR: https://github.com/OCA/helpdesk/pull/875

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdsk_mobile: Vue error when clicking the helpdesk button.

The v-text-field rendering the description when clicking the helpdesk button did cause a frontend error due to this props being given `:type="input_type"`.

As "input_type" is not declared anywhere we got this error:
`ReferenceError: input_type is not defined`

Moreover, the "type" props has been found not to be useful empirically (it is also not fund in Vuetify docs:  https://vuetifyjs.com/en/components/textareas/) so it has been removed.

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdsk_mobile: Vue error when clicking the helpdesk button.

The v-text-field rendering the description when clicking the helpdesk button did cause a frontend error due to this props being given `:type="input_type"`.

As "input_type" is not declared anywhere we got this error:
`ReferenceError: input_type is not defined`

Moreover, the "type" props has been found not to be useful empirically (it is also not fund in Vuetify docs:  https://vuetifyjs.com/en/components/textareas/) so it has been removed.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: validate description before ticket creation

The "Create Helpdesk Ticket" action previously allowed submission even
if the description was empty, leading to a 400 error printed on the page (in message pop up).

This change improves the user experience by:
- Adding a `:disabled` binding to the "Create Helpdesk Ticket" button
  based on the presence of the description text.
- Adding a Vuetify validation rule to the description text field to
  provide visual feedback to the user.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: validate description before ticket creation

The "Create Helpdesk Ticket" action previously allowed submission even
if the description was empty, leading to a 400 error printed on the page (in message pop up).

This change improves the user experience by:
- Adding a `:disabled` binding to the "Create Helpdesk Ticket" button
  based on the presence of the description text.
- Adding a Vuetify validation rule to the description text field to
  provide visual feedback to the user.

nicolas-delbovier-acsone commented:
Improved UI:

<img width="342" height="776" alt="image" src="https://github.com/user-attachments/assets/9eede20c-1565-45b4-a9e7-df186e5f0717" />

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: display helpdesk button in "set_quantity" screen

Displaying the button in "set_destination" screen felt a little too late. If there is a stock issue for example, it makes sense to let users create a ticket from the 'set_quantity' screen.

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: display helpdesk button in "set_quantity" screen

Displaying the button in "set_destination" screen felt a little too late. If there is a stock issue for example, it makes sense to let users create a ticket from the 'set_quantity' screen.

nicolas-delbovier-acsone commented:
For the moment, the helpdesk button shows after cancel and back buttons which feels a little odd:
<img width="395" height="689" alt="image" src="https://github.com/user-attachments/assets/13d3e82b-91ff-4f03-9b16-1cab0383522c" />

I spent quite some time attempting to position the button precisely, but the current method of direct string manipulation proved unreliable. Because the template contains multiple identical `<template>` tags, targeting the specific 'set_quantity' block via string searching is highly error-prone.

If anyone finds a better way to manipulate that template string, I'd be happy to hear about a solution 😀

nicolas-delbovier-acsone committed:
[IMP] run pre-commit

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk_mobile: add configure instructions

nicolas-delbovier-acsone committed:
[REF] shopfloor_reception_helpdesk: enable to create helpdesk tickets from any screen

nicolas-delbovier-acsone committed:
[REF] shopfloor_reception_helpdesk_mobile: add helpdesk button on other screens

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdesk: go back to "set_quantity" after creating a ticket from "set_quantity"

Previously, creating a helpdesk ticket from "set_quantity" state would make you flow to the "set_destination" state directly.

This does not make any sense since no qties will be inserted on the move line then. Moreover, there is no cancel button in "set_destination".

nicolas-delbovier-acsone commented:
I added the helpdesk button on 'select_move' and 'set_lot' screens since it makes sense depending on the type of issue you might have in the warehouse

nicolas-delbovier-acsone commented:
@jbaudoux

nicolas-delbovier-acsone committed:
[IMP] shopfloor_reception_helpdesk: add picking type config in shopfloor menu

To activate the helpdesk option in shopfloor, users need to allow helpdesk tickets on the picking type. It makes sense to allow to configure this option in the shopfloor menu directly instead of having to navigate to a different view

nicolas-delbovier-acsone committed:
[FIX] shopfloor_reception_helpdesk_mobile: replace "?." syntax for legacy webviews

jbaudoux commented:
/ocabot merge nobump

jbaudoux commented:
@sbidoul modules names to reserve on pypi shopfloor_reception_helpdesk, shopfloor_reception_helpdesk_mobile. Thanks in advance

sbidoul commented:
names reserved

jbaudoux commented:
/ocabot merge nobump
